diff --git a/src/paymentservice/index.js b/src/paymentservice/index.js
index 77f05162..c1e99325 100644
--- a/src/paymentservice/index.js
+++ b/src/paymentservice/index.js
@@ -17,6 +17,7 @@
 'use strict';
 
 const logger = require('./logger')
+const tracingEnabled = process.env.ENABLE_TRACING === "1";
 
 if (process.env.DISABLE_PROFILER) {
   logger.info("Profiler disabled.")
@@ -31,7 +32,7 @@ if (process.env.DISABLE_PROFILER) {
 }
 
 
-if (process.env.ENABLE_TRACING == "1") {
+if (tracingEnabled) {
   logger.info("Tracing enabled.")
 
   const { resourceFromAttributes } = require('@opentelemetry/resources');
