diff --git a/src/recommendationservice/recommendation_server.py b/src/recommendationservice/recommendation_server.py
index 7a8da7da..38cfee24 100755
--- a/src/recommendationservice/recommendation_server.py
+++ b/src/recommendationservice/recommendation_server.py
@@ -39,6 +39,7 @@ from opentelemetry.exporter.otlp.proto.grpc.trace_exporter import OTLPSpanExport
 
 from logger import getJSONLogger
 logger = getJSONLogger('recommendationservice-server')
+tracer = None
 
 def initStackdriverProfiling():
   project_id = None
@@ -68,6 +69,10 @@ def initStackdriverProfiling():
 
 class RecommendationService(demo_pb2_grpc.RecommendationServiceServicer):
     def ListRecommendations(self, request, context):
+        span = None
+        if tracer:
+            span = tracer.start_span("recommendation.logic")
+            span.set_attribute("product.count", len(request.product_ids))
         max_responses = 5
         # fetch list of products from product catalog stub
         cat_response = product_catalog_stub.ListProducts(demo_pb2.Empty())
@@ -83,6 +88,8 @@ class RecommendationService(demo_pb2_grpc.RecommendationServiceServicer):
         # build and return response
         response = demo_pb2.ListRecommendationsResponse()
         response.product_ids.extend(prod_list)
+        if span:
+            span.end()
         return response
 
     def Check(self, request, context):
@@ -122,6 +129,8 @@ if __name__ == "__main__":
             )
           )
         )
+        global tracer
+        tracer = trace.get_tracer("recommendationservice")
     except (KeyError, DefaultCredentialsError):
         logger.info("Tracing disabled.")
     except Exception as e:
